level_core/config/Adapter.py

```
@@ -0,0 +1,141 @@
+"""levelapp/config/interaction_request.py"""
+import os
+import json
+import yaml
+import logging
+
+from pydantic import BaseModel, HttpUrl, SecretStr, Field, computed_field
+from typing import Literal, Dict, Any, Callable, Optional
+from string import Template
+
+from dotenv import load_dotenv
+
+logger = logging.getLogger(__name__)
+load_dotenv()
+
+
+class EndpointConfig(BaseModel):
+    # Required
+    base_url: Optional[HttpUrl] = Field(default=os.getenv('ENDPOINT_URL', '') or None)
+    method: Literal["POST", "GET"] = Field(default="POST", description="endpoint method")
+
+    # Auth
+    api_key: SecretStr = Field(default=SecretStr(os.getenv('ENDPOINT_API_KEY', '')))
+    bearer_token: SecretStr = Field(default=SecretStr(os.getenv('BEARER_TOKEN', '')))
+    model_id: str = Field(default=os.getenv('MODEL_ID', ''))
+
+    # Data
+    payload_template: Dict[str, Any] = Field(default_factory=dict)
+    variables: Dict[str, Any] = Field(default_factory=dict)
+
+    @computed_field()
+    def full_url(self) -> str:
+        return str(self.base_url) if self.base_url else ""
+
+    @computed_field()
+    def headers(self) -> Dict[str, Any]:
+        headers: Dict[str, Any] = {"Content-Type": "application/json"}
+        if self.model_id:
+            headers["x-api-key"] = self.model_id
+        if self.bearer_token:
+            headers["Authorization"] = f"Bearer {self.bearer_token.get_secret_value()}"
+        if self.api_key:
+            headers["x-api-key"] = self.api_key.get_secret_value()
+        return headers
+
+    @computed_field
+    def payload(self) -> Dict[str, Any]:
+        """Return fully prepared payload depending on template or full payload."""
```

add `load_template()` after line 48. and raise a `ValueError ` after the if case instead of returning and empty payload template.